app/views/PickerView.tsx

```diff
@@ -81,16 +89,31 @@ class PickerView extends React.PureComponent<IPickerViewProps, IPickerViewState>
 	};
 
 	onChangeText = debounce(
-		async (text: string) => {
+		async (searchText: string) => {
 			if (this.onSearch) {
-				const data = await this.onSearch(text);
-				this.setState({ data });
+				const data = await this.onSearch(searchText);
+				if (data?.data) {
+					this.setState({ ...data, searchText });
+				}
 			}
 		},
 		300,
 		true
 	);
 
+	onEndReached = async () => {
+		const { route } = this.props;
+		const { data, offset, total, searchText } = this.state;
+		const onEndReached = route.params?.onEndReached;
+		const newOffset = offset + this.count;
```

Comment:
Isn't `offset + this.count` the same as `data.length`?
I think it's confusing to receive `count` as a prop on `PickerView` and limiting as a generic component.
It shouldn't have to know `count` is the value we're sending as param for a http request.